runtime/local/src/lib.rs

```diff
@@ -528,12 +528,28 @@ parameter_types! {
     pub WeightPerGas: Weight = Weight::from_ref_time(WEIGHT_PER_GAS);
 }
 
+/// Ensure that the origin is EVM compatible.
+pub struct EnsureEvmOrigin;
+impl<OuterOrigin> pallet_evm::EnsureAddressOrigin<OuterOrigin> for EnsureEvmOrigin
+where
+    OuterOrigin: Into<Result<pallet_account::NativeAndEVM, OuterOrigin>>
+        + From<pallet_account::NativeAndEVM>,
+{
+    type Success = ();
+    fn try_address_origin(address: &H160, origin: OuterOrigin) -> Result<(), OuterOrigin> {
+        origin.into().and_then(|o| match o {
+            pallet_account::NativeAndEVM::H160(a) if *address == a => Ok(()),
+            r => Err(OuterOrigin::from(r)),
+        })
+    }
+}
+
 impl pallet_evm::Config for Runtime {
     type FeeCalculator = BaseFee;
     type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
     type WeightPerGas = WeightPerGas;
     type BlockHashMapping = pallet_ethereum::EthereumBlockHashMapping<Runtime>;
-    type CallOrigin = pallet_evm::EnsureAddressRoot<AccountId>;
+    type CallOrigin = EnsureEvmOrigin;
```

Comment:
This should remain `root` - it's only related to who can call the actual `pallet-evm` extrinsics (which should be no one), it won't help with signature verification in `pallet-ethereum`.

The whole point of switching away from `pallet-evm` is to still emit relevant events for EVM indexers. It's all described in the BL ticket.

Reply:
It's not, details are in the backlog item.